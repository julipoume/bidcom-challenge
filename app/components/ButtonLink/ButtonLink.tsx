import React from "react";
import Link from "next/link";
import clsx from "clsx";

interface ButtonLinkProps {
  link: string;
  title: string;
  showIn: "mobile" | "desktop" | "both";
}

export default function ButtonLink({ link, title, showIn }: ButtonLinkProps) {
  return (
    <Link
      href={link}
      className={clsx(
        "bg-black text-white rounded-[10px] flex items-center gap-2",
        {
          "block md:hidden w-full justify-center py-[17px] px-8": showIn === "mobile",
          "hidden md:flex w-fit justify-start py-[15px] px-8": showIn === "desktop",
          "block w-full md:w-fit justify-center md:justify-start px-8 py-[17px] md:py-[15px]": showIn === "both",
        }
      )}
    >
      <span className="text-sm md:text-base">{title}</span>
      <img
        src="/assets/icons/right-arrow.png"
        alt="Arrow"
        className="ml-2 w-4 h-4"
      />
    </Link>
  );
}
