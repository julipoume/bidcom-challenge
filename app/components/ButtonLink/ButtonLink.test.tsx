import React from "react";
import { render, screen } from "@testing-library/react";
import ButtonLink from "./ButtonLink";

describe("ButtonLink component", () => {
  it("snapshot", () => {
    const { container } = render(
      <ButtonLink link="/test" title="Desktop Link" showIn="desktop" />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correctly with mobile props", () => {
    render(<ButtonLink link="/test" title="Click me" showIn="mobile" />);

    const link = screen.getByRole("link", { name: /click me/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveClass("block");
    expect(link).toHaveClass("md:hidden");
  });

  it("renders correctly with desktop props", () => {
    render(<ButtonLink link="/test" title="Desktop Link" showIn="desktop" />);

    const link = screen.getByRole("link", { name: /desktop link/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveClass("hidden");
    expect(link).toHaveClass("md:flex");
  });

  it("renders the right arrow icon", () => {
    render(<ButtonLink link="/test" title="Icon test" showIn="both" />);

    const img = screen.getByAltText("Arrow");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "/assets/icons/right-arrow.png");
  });
});
