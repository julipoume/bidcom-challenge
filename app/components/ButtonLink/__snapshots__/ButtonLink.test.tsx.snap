// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ButtonLink component snapshot 1`] = `
<div>
  <a
    class="bg-black text-white rounded-[10px] flex items-center gap-2 hidden md:flex w-fit justify-start py-[15px] px-8"
    href="/test"
  >
    <span
      class="text-sm md:text-base"
    >
      Desktop Link
    </span>
    <img
      alt="Arrow"
      class="ml-2 w-4 h-4"
      src="/assets/icons/right-arrow.png"
    />
  </a>
</div>
`;
